sdks/java/io/clickhouse/src/main/java/org/apache/beam/sdk/io/clickhouse/TableSchema.java

```diff
@@ -111,6 +112,14 @@ public static Schema.FieldType getEquivalentFieldType(ColumnType columnType) {
         return Schema.FieldType.STRING;
       case BOOL:
         return Schema.FieldType.BOOLEAN;
+      case TUPLE:
+        List<Schema.Field> fields =
+            columnType.tupleTypes().entrySet().stream()
+                .map(x -> Schema.Field.of(x.getKey(), Schema.FieldType.DATETIME))
```

Comment:
Hi @mzitnik, 
Shouldn't each key inside a tuple be mapped to their respective data type ? Any reason for being `DATETIME`?

Reply:
This quick adaptation seems to work: 
```
case TUPLE:
                List<TableSchema.Column> columns = columnType.tupleTypes().entrySet().stream().map(
                                x -> TableSchema.Column.of(x.getKey(), x.getValue())
                        ).collect(Collectors.toList());
                TableSchema tupleSchema = TableSchema.of(columns.toArray(new TableSchema.Column[0]));
                Schema tupleAsRowSchema = getEquivalentSchema(tupleSchema);
                return Schema.FieldType.row(tupleAsRowSchema);
```

Reply:
Hi, 

Also, it seems that `Array(Tuple(*))` is not supported when calling `ClickHouseIO.getTableSchema`. 

This seems to fix the issue : 
`
if (type.toLowerCase().trim().startsWith("tuple(")) {
     String content = tuplePreprocessing(type);
     columnType = TableSchema.ColumnType.parse(content);
}
else if (type.toLowerCase().trim().startsWith("array(tuple(")) {
     String content = tuplePreprocessing(type);
     columnType = TableSchema.ColumnType.parse(content);
}
`

Reply:
@erube and @wattache, thanks for the feedback and change